portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +34,74 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> messages;
+
+  public void init() {
+      //submitTime = new Date();
+      messages = new ArrayList<String>();
+  }
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+  throws IOException {
+
+      Query query = new Query("Task//");//.addSort("timestamp", SortDirection.DESCENDING);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+    
+      for (Entity entity : results.asIterable()) {
+          long id = entity.getKey().getId();
+          String message = (String) entity.getProperty("message");
+          messages.add(message);
+      }
+
+      response.setContentType("application/json;");
+      response.getWriter().println(convertToJsonUsingGson(messages));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String clientMessage = request.getParameter("clientMessage");
+      long timeStamp = System.currentTimeMillis();
+
+      /*System.out.println(htmlMessage);
+      messages.add(htmlMessage);
+      Gson gson = new Gson();
+      String json = convertToJson(clientMessage));*/
+      Entity messageEntity = new Entity("Messages");
+      messageEntity.setProperty("message", clientMessage);
+      messageEntity.setProperty("timestamp", timeStamp);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(messageEntity);
+
+      response.sendRedirect("/index.html");
+
+    }
+
+  private String convertToJsonUsingGson(ArrayList<String> messageList) {
+      Gson gson = new Gson();
+      String json = gson.toJson(messageList);
+    
+      return json;
+  }
+
+  private String convertToJson(ArrayList<String> messageList) {
```

Comment:
Is this function needed any more?

Reply:
The reason I kept this function is in case I want to try sorting the comments by more than just the content (timestamp or name) and I'm not sure Gson can convert different data types like ints into Json. I'll do more research on whether that's correct or not.